pkg/bytecode/compiler.go

```diff
@@ -31,6 +31,9 @@ type Compiler struct {
 	constants    []value
 	instructions Instructions
 	globals      *SymbolTable
+	// breakStack tracks the positions of all break statements
+	// in the bytecode
+	breakStack []int
```

Comment:
I'm interested if there are any suggestions on how to better handle this. A break can be nested an arbitrary number of nodes deep inside a loop so some sort of compiler-level state is needed to track where to backfill.

Reply:
I would not do this as a stack, but just a list. At the start of a function that compiles a loop (currently just `compileWhileStatement()`, but I guess the upcoming `compileForStatement()`), you save the current "break list", empty the one in the compiler state and then compile the block that is the statement. After that, range over the break list and patch all the break statements, and then restore the break list you saved at the top.

This is a bit simpler than the current method of having all ancestor loops in the stack and identifying where the nesting ends.

Alternatively, make this stack a stack of slices. Push a new empty slice onto the stack, have `break` add to that list on the top of the stack, and at the end of the loop compile function, pop that slice off and iterate it patching all of the breaks in that list.

You'll need something similar for `continue` - you don't need a slice, just a single continue IP for generating the jump, but you will either need a stack for that too or save it like my original suggestion.

Perhaps combine these into a loop state stack:
```
type loopState struct {
  breakInstructions []int
  continueIP int
}
```

and push a fresh one onto the stack each loop compilation.

Reply:
Good thinking about the break list, that's much simpler.

> You'll need something similar for continue

The [evy spec](https://github.com/evylang/evy/blob/main/docs/spec.md) doesn't mention `continue` and with a quick test on the website I confirmed that it isn't implemented.

Reply:
Ooops. No `continue`. My mistake.